### PR TITLE
zynq7000-spi: add support for external SS line control

### PR DESCRIPTION
**Pull request by Maxez**
## Description

Allow for external SS control.
Changed SPI registers from enums to defines for MISRA conformance.

## Motivation and Context

[JIRA: PP-44](https://jira.phoenix-rtos.com/browse/PP-44)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: armv7a9-zynq7000-zedboard

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: spi/zynq7000-spi/libzynq7000-spi.c [hunk 1]
```diff
@@ -386,13 +401,18 @@ static int spi_initPins(spi_t *spi)
...
 	for (i = 0; i < sizeof(spi->pins) / sizeof(spi->pins[0]); i++) {
+		/* Skip not configured pins */
+		if (spi->pins[i].pin < 0) {
+			continue;
+		}
+
 		err = spi_setPin(spi->pins[i].pin, spi->pins[i].cfg);
 		if (err < 0) {
-			break;
+			return err;
```
Comment (agkaminski) [inline spi/zynq7000-spi/libzynq7000-spi.c:411]: Why this change?

Comment (Maxez) [reply spi/zynq7000-spi/libzynq7000-spi.c:411]: If no SPI pin is to be configured (`spi->pins[i].pin < 0` for all pins) than the function returned uninitialzed `err`. Added this change to fix that issue.

File: spi/zynq7000-spi/libzynq7000-spi.c [hunk 2]
```diff
@@ -259,15 +270,23 @@ int spi_setSpeed(unsigned int dev, unsigned int speed)
...
 	unsigned int cfg;
 
-	if (ss < 0) {
+	/* Check for external SS control */
+	if (ss == SPI_SS_EXTERNAL) {
+		return;
+	}
+
+	/* Raise SS line high */
+	if (state) {
```
[the remark this answers is not in the document]
Comment (Maxez) [reply spi/zynq7000-spi/libzynq7000-spi.c:283]: Changed to `state > 0` (seems better to me).

File: spi/zynq7000-spi/libzynq7000-spi.c [hunk 3]
```diff
@@ -296,7 +315,7 @@ int spi_xfer(unsigned int dev, unsigned int ss, const void *out, size_t olen, vo
 		return -ENODEV;
 	}
 
-	if ((ss >= SPI_SS_COUNT) || (spi->pins[ss].pin < 0)) {
+	if ((ss != SPI_SS_EXTERNAL) && ((ss >= SPI_SS_COUNT) || (spi->pins[ss].pin < 0))) {
```
[the remark this answers is not in the document]
Comment (Maxez) [reply spi/zynq7000-spi/libzynq7000-spi.c:318]: Not sure what's the issue here. Can you be more specific?

[the remark this answers is not in the document]
Comment (agkaminski) [reply spi/zynq7000-spi/libzynq7000-spi.c:318]: You don't check if ss is greater or equal zero - if it cannot be less than zero (apart from SPI_SS_EXTERNAL) then it's ok

[the remark this answers is not in the document]
Comment (Maxez) [reply spi/zynq7000-spi/libzynq7000-spi.c:318]: Oh, I get it now. Thanks for the explanation :)
`ss` can't be less than zero (it has `unsigned int` type). `SPI_SS_EXTERNAL` is the only special value, which isn't really negative - it's max unsigned int.

[the remark this answers is not in the document]
Comment (agkaminski) [reply spi/zynq7000-spi/libzynq7000-spi.c:318]: Oh, I don't know why I thought that it was int. This is signed/unsigned comparison, but for equal/unequal this shouldn't matter.

**Review by agkaminski — APPROVED** [review]
